ledger/metrics/src/main/scala/com/daml/metrics/Metrics.scala

```diff
@@ -4,71 +4,81 @@
 package com.daml.metrics
 
 import com.codahale.metrics.MetricRegistry
+import com.daml.metrics.api.MetricHandle.{LabeledMetricsFactory, MetricsFactory}
 import com.daml.metrics.api.MetricName
 import com.daml.metrics.api.dropwizard.DropwizardMetricsFactory
-import com.daml.metrics.api.opentelemetry.OpenTelemetryFactory
+import com.daml.metrics.api.opentelemetry.OpenTelemetryMetricsFactory
 import com.daml.metrics.grpc.DamlGrpcServerMetrics
 import io.opentelemetry.api.GlobalOpenTelemetry
 import io.opentelemetry.api.metrics.Meter
 
 object Metrics {
 
   def apply(registry: MetricRegistry, otelMeter: Meter) =
-    new Metrics(new DropwizardMetricsFactory(registry), new OpenTelemetryFactory(otelMeter))
-
-  lazy val ForTesting = new Metrics(
-    new DropwizardMetricsFactory(new MetricRegistry),
-    new OpenTelemetryFactory(GlobalOpenTelemetry.getMeter("test")),
-  )
+    new Metrics(
+      new DropwizardMetricsFactory(registry),
+      new OpenTelemetryMetricsFactory(otelMeter),
+      registry,
+    )
+
+  lazy val ForTesting: Metrics = {
+    val registry = new MetricRegistry
+    new Metrics(
+      new DropwizardMetricsFactory(registry),
+      new OpenTelemetryMetricsFactory(GlobalOpenTelemetry.getMeter("test")),
+      registry,
+    )
+  }
 }
 
 final class Metrics(
-    val dropwizardFactory: DropwizardMetricsFactory,
-    val openTelemetryFactory: OpenTelemetryFactory,
+    val defaultMetricsFactory: MetricsFactory,
+    val labelMetricsFactory: LabeledMetricsFactory,
+    val registry: MetricRegistry,
```

Comment:
Using `MetricsFactory` and `LabeledMetricsFactory` removes the direct dependency on Dropwizard and OpenTelemetry.
But `registry` adds back the dependency on Dropwizard.
I assume resolving those usage of Dropwizard are part of other tasks.

Reply:
Yes, we have to refactor/remove a few more dropwizard components before we can fully remove it.
We could've removed it from the metrics object but overall it would've added too much noise with little benefit.